### PR TITLE
emit github events

### PR DESCRIPTION
**Pull request by williamkapke**
<del>Not tested- but should mostly be correct ;)</del>
EDIT: Testing & working!

I'm using a `.env` file like discussed in https://github.com/nodejs-github-bot/github-bot/pull/7#issuecomment-205525555

<del>You'll probably notice that `github-webhook-handler` is referencing my fork- hopefully @rvagg will [accept my PR](https://github.com/rvagg/github-webhook-handler/pull/19) and that can be replaced soon. If not, I can do it a uglier way.</del> EDIT: Needed to go a different direction in the end anyhow.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 node_modules
 npm-debug.log
 
-.DS_Store
\ No newline at end of file
+.DS_Store
+.env
```
Comment (Fishrock123) [inline .gitignore]: metadata LGTM, but can you move it to a separate PR? we can probably merge it immediately

File: package.json [hunk 2]
```diff
@@ -13,8 +13,11 @@
   "license": "MIT",
   "dependencies": {
     "body-parser": "^1.15.0",
+    "debug": "^2.2.0",
+    "dotenv": "^2.0.0",
     "express": "^4.13.4",
     "github": "^0.2.4",
+    "github-webhook-handler": "https://github.com/williamkapke/github-webhook-handler#9ca8d8103201a60fb80b4ec12fee88abd3f59692",
```
Comment (Fishrock123) [inline package.json]: `williamkapke/github-webhook-handler#9ca8d81`

**Comment by williamkapke** [comment]
This just ran this successfully here: https://github.com/no-duh/nodejs.org/pull/1

Travis updated the PR status also... I'm guessing it is because I gave it permission to and the https://github.com/node/nodejs.org repo doesn't give Travis the permission to..?

I tried to follow along https://github.com/nodejs/nodejs.org/issues/355 but didn't grasp what is going on. I'd like to completely mirror the behavior if I can.

**Comment by Fishrock123** [comment]
@williamkapke Try doing it in my org, I can set the permissions up.

You need to enable advanced restrictions, try to sync travis, ask from permission from your own permissions, turn travis on after you;ve approved it, and then revoke the permission from the org.

**Comment by williamkapke** [comment]
Ok- as you probably saw, I created the `nodejs.org` repo there, but then deleted it. I didn't use it because it added everyone as watchers and I wanted to be considerate about not spamming. I'll set it up now that I know it is working and the notifications shouldn't be as bad.

**Comment by williamkapke** [comment]
@Fishrock123 ok, I guess you'll need to setup Travis. It won't let me do it...

screenshot 2016-04-06 20 21 31

File: scripts/display-travis-status.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+'use strict'
+
+const debug = require('debug')('display_travis_status')
+const pollTravis = require('../lib/pollTravis')
+
+module.exports = function (app) {
+  app.on('pull_request.opened', (event) => {
+    const owner = event.repository.owner.login
+    const repo = event.repository.name
+    if (repo !== 'nodejs.org') return
```
Comment (phillipj) [inline scripts/display-travis-status.js]: We want to run this on multiple repos. citgm was enabled yesterday, and readable-stream will also get enabled as soon as #12 gets merged.

Comment (williamkapke) [inline scripts/display-travis-status.js]: Yup, I wanted to make sure this was focused on what it was written for when I started touching it. I figured that was _probably_ the case... but I'm not directly involved with that, so I leaned towards sheltering it from other repos to start.

I can add additional repo names now if you want... or it can come in another PR. LMK!

Comment (phillipj) [inline scripts/display-travis-status.js]: I was leaning more against removing that check all together. It isn't there today, and we haven't had the need to add either.

Comment (williamkapke) [inline scripts/display-travis-status.js]: The bot will be used on repos that do not have builds (like TSC) or do not use Travis (like `node`).

Comment (williamkapke) [inline scripts/display-travis-status.js]: Oh, and my understanding is that there is _1 bot to rule them all!_... but obviously that might not end up being true.

Comment (phillipj) [inline scripts/display-travis-status.js]: > The bot will be used on repos that do not have builds (like  TSC ) or do not use Travis (like  node ).

Ahhhh good point! Atleast allow citgm and readable-stream then. For later we could check if the repo in question uses Travis, but I'm fine with adding that later in another PR.

Comment (jbergstroem) [inline scripts/display-travis-status.js]: I agree with the sentiment of having one bot to rule them all. It will raise requirements for the software (needs to store state and likely other stuff) as well as security surrounding it since it will contain access keys to github and jenkins. If the bot turns up being adopted by nodejs org (has it already?) the build team is happy to arrange deployment.

Comment (phillipj) [inline scripts/display-travis-status.js]: > the build team is happy to arrange deployment.

Awesome :+1:

Comment (jbergstroem) [inline scripts/display-travis-status.js]: Just to make one thing clear; I'll bring it up on a meeting and see if there are objections, but since we've been talking about writing software that polls CI and pings back for a year I don't see how it wouldn't be accepted. 

I'll put together my thoughts on how this needs to work which may or may not lead to a PR.

**Comment by Fishrock123** [comment]
@williamkapke Ok https://travis-ci.org/TestOrgPleaseIgnore/nodejs.org is "semi-on" now.

**Comment by phillipj** [comment]
@williamkapke merged #12 to get enable the bot on the readable-stream repo, the merge conflict should be trivial to resolve. I renamed `pollTravis.pollAndComment()` -> `pollTravis.pollThenComment()`.

By rebasing on top of master you'll even get Travis to run this PR ;)

Please confirm this is good to go after the conflict has been resolved, and I'll be more than happy to push it to production.

**Comment by Fishrock123** [comment]
`pollThenComment` should probably renamed to `pollThenStatus` here I guess?

**Comment by williamkapke** [comment]
Rebased. Tested. Here's the log output...
```
event@TestOrgPleaseIgnore/nodejs.org: pull_request.opened
Thu, 07 Apr 2016 19:48:28 GMT display_travis_status /TestOrgPleaseIgnore/nodejs.org/pull/2 opened
! TestOrgPleaseIgnore/nodejs.org/#2 Was not able to find matching build, will do check #2 in 30 seconds
* TestOrgPleaseIgnore/nodejs.org/#2 "started" build found, will do check #3 in 30 seconds
* TestOrgPleaseIgnore/nodejs.org/#2 "started" build found, will do check #4 in 30 seconds
* TestOrgPleaseIgnore/nodejs.org/#2 Github comment created
event@TestOrgPleaseIgnore/nodejs.org: issue_comment.created
```

**Comment by phillipj** [comment]
> `pollThenComment` should probably renamed to `pollThenStatus` here I guess?

The latter sounds like we're using the Status API, but we're still commenting on PR's here.. Or did I misunderstand?

**Comment by Fishrock123** [comment]
The status here was emitted by travis itself, right?

**Comment by phillipj** [comment]
LGTM

Looks like it introduces secrets, so we'll have to update the active PRs before pushing it to prod

**Comment by williamkapke** [comment]
Ah yes. I added the header validation. So the `GITHUB_WEBHOOK_SECRET ` will need to be added.

File: scripts/display-travis-status.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+const debug = require('debug')('display_travis_status')
+const pollTravis = require('../lib/pollTravis')
+const enabledRepos = ['citgm', 'readable-stream', 'nodejs.org']
+
+module.exports = function (app) {
+  app.on('pull_request.opened', (event) => {
+    const owner = event.repository.owner.login
+    const repo = event.repository.name
+    if (!~enabledRepos.indexOf(repo)) return
```
Comment (Fishrock123) [inline scripts/display-travis-status.js]: fancy but hard to read ;)

Comment (williamkapke) [inline scripts/display-travis-status.js]: this WAS `enabledRepos.includes(repo)`.. but then I realized I'm getting too comfortable with the nighty version and it isn't available in v4/5.. so I switched it to the old hackery way.

Comment (Fishrock123) [inline scripts/display-travis-status.js]: Oh I just meant `indexOf === -1`, or `!==` is a lot clearer.
